assets/generator/directory/main.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type Metadata struct {
+	Name     string `json:"name"`
+	Platform string `json:"platform"`
+	Provider string `json:"provider"`
+}
+
+// Function to get user input with a default value
+func getInput(prompt, defaultValue string) string {
+	fmt.Printf("%s (default: %s): ", prompt, defaultValue)
+	reader := bufio.NewReader(os.Stdin)
+	input, _ := reader.ReadString('\n')
+	input = strings.TrimSpace(input)
+	if input == "" {
+		return defaultValue
+	}
+	return input
+}
+
+func main() {
+	// Set default values
+	defaultRepoPath := "~/dev/kics"
+	defaultPlatform := "terraform"
+	defaultProvider := "aws"
+
+	// Get user input with default values
+	repoPath := getInput("Enter the path to the repo", defaultRepoPath)
+	platform := getInput("Enter the platform", defaultPlatform)
+	provider := getInput("Enter the provider", defaultProvider)
+	ruleName := getInput("Enter the rule name", "")
+	if ruleName == "" {
+		fmt.Println("Rule name is required. Exiting.")
+		return
```

Comment:
suggestion: set exit code to non-zero value on error.
```suggestion
		os.Exit(1)
```